src/details/ArborX_MinimumSpanningTree.hpp

```diff
@@ -86,12 +87,21 @@ struct FindComponentNearestNeighbors
       , _metric{metric}
       , _radii(radii)
       , _lower_bounds(lower_bounds)
+      , _found_edges(Kokkos::view_alloc(space, Kokkos::WithoutInitializing,
+                                        "ArborX::MST::found_edges"),
+                     bvh.size())
+
   {
     int const n = bvh.size();
     ARBORX_ASSERT(labels.extent_int(0) == 2 * n - 1);
     ARBORX_ASSERT(edges.extent_int(0) == n);
     ARBORX_ASSERT(radii.extent_int(0) == n);
 
+    constexpr auto inf = KokkosExt::ArithmeticTraits::infinity<float>::value;
+    WeightedEdge uninitialized_edge{INT_MAX, INT_MAX, inf};
+    Kokkos::deep_copy(space, _edges, uninitialized_edge);
+    Kokkos::deep_copy(space, _found_edges, uninitialized_edge);
```

Comment:
Because individual entries for `_edges` and `_found_edges` are used in the `atomic_min` context, each field is now initialized to its maximum value.